scripts/bump.sh

```diff
@@ -113,45 +122,154 @@ check_iscsc_remote () {
+	[ -z "$DRY_RUN" ] && git switch -c ${bump_branch}
+}
 
 # -------------------------------- Version Bump --------------------------------
 # ------------------------- Bump frontend and backend --------------------------
 
-log_info 'Bumping `frontend`'
-(
-	cd frontend
-	[ -z "$DRY_RUN" ] && npm version "${NEW_VERSION}" --no-git-tag-version
-)
-log_info 'Bumping `backend`'
-(
-	cd backend
-	[ -z "$DRY_RUN" ] && npm version "${NEW_VERSION}" --no-git-tag-version
-)
+bump_modules () {
+	local new_version="$1"
+	log_info 'Bumping `frontend`'
+	(
+		cd frontend
+		[ -z "$DRY_RUN" ] && npm version "${new_version}" --no-git-tag-version
+	)
+	log_info 'Bumping `backend`'
+	(
+		cd backend
+		[ -z "$DRY_RUN" ] && npm version "${new_version}" --no-git-tag-version
+	)
+}
 
 # ----------------------------- Bump root and push -----------------------------
 
-log_info 'Bumping `root`'
-[ -z "$DRY_RUN" ] && { npm version "${NEW_VERSION}" --no-git-tag-version || exit 1; }
-[ -z "$DRY_RUN" ] && { git commit -m "Bump to version ${NEW_VERSION}" || exit 1; }
+bump_root () {
+	local new_version="$1"
+	log_info 'Bumping `root`'
+	[ -z "$DRY_RUN" ] && { npm version "${new_version}" --no-git-tag-version || exit 1; }
+	[ -z "$DRY_RUN" ] && { git commit -m "Bump to version ${new_version}" || exit 1; }
+}
+
+# -- Help and usage --
+
+usage () {
+	echo "Usage: $USAGE"
+	echo "Type -h or --help for the full help."
+	echo "See https://github.com/iScsc/iscsc.fr/wiki/Version-bump-procedure#automatic-version-bump for full documentation"
+	exit 0
+}
+
+help () {
+	echo "bump.sh:"
+	echo "     a script to easily bump the version of the iScsc website project."
+	echo "     it should guide you when you make mistakes, e.g. providing less or more than 1 argument."
+	echo ""
+	echo "Usage:"
+	echo "     $USAGE"
+	echo ""
+	echo "Options:"
+	echo "     -h/--help                shows this help."
+	echo "     -p/--patch               bump to patch version (1.2.3 --> 1.2.4)."
+	echo "     -m/--minor               bump to minor version (1.2.3 --> 1.3.0)."
+	echo "     -M/--major               bump to major version (1.2.3 --> 2.0.0)."
+	exit 0
+}
+
+# -- Version Bump Functions --
+
+bump_major () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+
+    echo "$((major+1)).0.0"
+}
+
+
+bump_minor () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+    minor=${bits[1]}
+
+    echo "$major.$((minor+1)).0"
+}
+
+
+bump_patch () {
+    bits=($(break_down_version $1))
+    major=${bits[0]}
+    minor=${bits[1]}
+    patch=${bits[2]}
+
+    echo "$major.$minor.$((patch+1))"
+}
+
+# -------------------------------- Main Section --------------------------------
+
+main () {
+	# Run all basic checks
+	check_pwd
+	check_arg "${NB_ARGS}"
+	check_dependencies
+	check_clean_git_working_dir
+
+	# Define version variables
+	NEW_VERSION=""
+	CURRENT_VERSION=$(npm pkg get version | sed 's/"//g')
+
+	# Extract new version from arguments
+	while [[ $# -gt 0 ]]; do
```

Comment:
here i would fully parse the arguments before doing anything!

for instance, running `./script/bump.sh --help` does run `npm` and stuff happens...
this is not what i expect for a `--help` :thinking: 

what i would do is
- put the `while` at the top of the `main`
- instead of running `NEW_VERSION=$(bump...)` you can store the action to be performed
- you execute `check_pwd`, etc, ...
- finally you switch over the action value and run the appropriate `bump_*`

Reply:
that's a really good remark I'll change that! tks!

Reply:
done!